tests/security/apparmor_profile/usr_sbin_dnsmasq.pm

```diff
@@ -0,0 +1,95 @@
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Package: apparmor-utils nscd
+# Summary: Test with "usr.sbin.dnsmasq" is in "enforce" mode and AppArmor is
+#          "enabled && active", libvirt manages the NAT network depending on
+#          dnsmasq have no error
+# - Create virtual machines with libvirt in NAT network setting
+# - Switch dnsmasq to enforce mode
+# - Check network status of the guest machine(s)
+# - Check audit.log for errors related to dnsmasq
+# Maintainer: llzhao <llzhao@suse.com>
+# Tags: poo#103458, tc#1767545
+
+use base 'apparmortest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub check_audit_log {
+    my ($self) = shift;
+    my $log_file = $apparmortest::audit_log;
+
+    # Check audit log contains no related error
+    my $script_output = script_output "cat $log_file";
+    if ($script_output =~ m/type=AVC .*apparmor=.*DENIED.* profile=.*dnsmasq.* comm=.*libvirt.*/sx) {
+        record_info('ERROR', "There are errors found in $log_file", result => 'fail');
+        $self->result('fail');
+    }
+    record_info("audit log:", "$script_output");
+}
+
+sub run {
+    my ($self) = shift;
+    my $log_file = $apparmortest::audit_log;
+    my $result_file = '/tmp/result_file';
+    my $image_path = '/var/lib/libvirt/images';
+
+    select_console 'root-console';
+
+    # Check the version for "SLE-20353 QA: dnsmasq update to 2.83"
+    my $current_ver = script_output('zypper se -s dnsmasq libnettle');
```

Comment:
@lilyeyes 
I have 2 questions here
- You dump the dnsmasq version from the scc/proxyscc rather than the version which is installed into the system, right?
- Do you need to install the dnsmasq into the system? looks I did not find it from the code or if I missed something?

Reply:
Good question:
1. yes, but I only aim to show the available versions and then do manually check.
2. good pointing, when installing "libvirt" the "dnsmasq" will be installed automatically. I will revise code to avoid the confusion.